load-testing/tests/relays_stress_test.go

```diff
@@ -0,0 +1,457 @@
+		}
+		for _, app := range s.preparedApplications {
+			s.txContext.GetKeyring().Delete(app.keyName)
+		}
+	})
+
+	// Initialize the provisioned gateway and suppliers keyName->URL map that will
+	// be populated from the load test manifest.
+	s.gatewayUrls = make(map[string]string)
+	s.suppliersUrls = make(map[string]string)
+
+	// Set up the blockClient that will be notifying the suite about the committed blocks.
+	s.blockClient = testblock.NewLocalnetClient(s.ctx, s.TestingT.(*testing.T))
+	channel.ForEach(
+		s.ctx,
+		s.blockClient.CommittedBlocksSequence(s.ctx),
+		func(ctx context.Context, block client.Block) {
+			s.latestBlock = block
+		},
+	)
+
+	// Setup the txContext that will be used to send transactions to the network.
+	s.txContext = testtx.NewLocalnetContext(s.TestingT.(*testing.T))
+
+	// Get the relay cost from the tokenomics module.
+	s.relayCost = s.getRelayCost()
+
+	// Setup the tx listener for on-chain events to check and assert on transactions results.
+	s.setupTxEventListeners()
+
+	// Initialize the funding account.
+	s.initFundingAccount(fundingAccountKeyName)
+
+	// Initialize the provisioned gateways and suppliers from the load test manifest.
+	s.initializeProvisionedActors()
+
+	// Some suppliers may already be staked at genesis, ensure that staking during
+	// this test succeeds by increasing the sake amount.
+	minStakeAmount := s.getProvisionedActorsCurrentStakedAmount()
+	stakeAmount = sdk.NewCoin("upokt", math.NewInt(minStakeAmount+1))
+}
+
+func (s *relaysSuite) ARateOfRelayRequestsPerSecondIsSentPerApplication(appRPS string) {
+	relayRatePerApp, err := strconv.ParseInt(appRPS, 10, 32)
+	require.NoError(s, err)
+
+	s.relayRatePerApp = relayRatePerApp
+}
+
+func (s *relaysSuite) TheFollowingInitialActorsAreStaked(table gocuke.DataTable) {
+	// Store the initial counts of the actors to be staked to be used later in the test,
+	// when information about max actors to be staked is available.
+	s.supplierInitialCount = table.Cell(3, 1).Int64()
+	s.gatewayInitialCount = table.Cell(1, 1).Int64()
+	s.appInitialCount = table.Cell(2, 1).Int64()
+}
+
+func (s *relaysSuite) MoreActorsAreStakedAsFollows(table gocuke.DataTable) {
+	plan := actorPlans{
+		gateways: actorPlan{
+			initialAmount:   s.gatewayInitialCount,
+			incrementAmount: table.Cell(1, 1).Int64(),
+			incrementRate:   table.Cell(1, 2).Int64(),
+			maxAmount:       table.Cell(1, 3).Int64(),
+		},
+		apps: actorPlan{
+			initialAmount:   s.appInitialCount,
+			incrementAmount: table.Cell(2, 1).Int64(),
+			incrementRate:   table.Cell(2, 2).Int64(),
+			maxAmount:       table.Cell(2, 3).Int64(),
+		},
+		suppliers: actorPlan{
+			initialAmount:   s.supplierInitialCount,
+			incrementAmount: table.Cell(3, 1).Int64(),
+			incrementRate:   table.Cell(3, 2).Int64(),
+			maxAmount:       table.Cell(3, 3).Int64(),
+		},
+	}
+
+	s.validateActorPlans(&plan)
+
+	// The test duration is the longest duration of the three actor increments.
```

Comment:
All of these comments are fantastic

Reply:
It's been a joint effort (:heart: @red-0ne)